plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/HiddenAction.java

```diff
@@ -45,21 +48,34 @@ public HiddenAction(Object viewer, IFeatureModel featureModel) {
 
 	@Override
 	public void run() {
-		setChecked(feature.getStructure().isHidden());
-		final SetFeatureToHiddenOperation op = new SetFeatureToHiddenOperation(feature, featureModel);
-
+		changeHiddenStatus(isEveryFeatureHidden());
+		setChecked(isEveryFeatureHidden());
+	}
+	
+	private boolean isEveryFeatureHidden() {
+		for (IFeature tempFeature : featureArray) {
+			if (!(tempFeature.getStructure().isHidden())) {
+				return false;
+			}
+		}
+		return true;
+	}
+	
+	private void changeHiddenStatus(boolean allHidden) {
+		final SetFeatureToHiddenOperation op = 
+				new SetFeatureToHiddenOperation(featureModel, allHidden, getSelectedFeatures());
 		try {
 			PlatformUI.getWorkbench().getOperationSupport().getOperationHistory().execute(op, null, null);
 		} catch (final ExecutionException e) {
 			FMUIPlugin.getDefault().logError(e);
-
 		}
 	}
 
 	@Override
 	protected void updateProperties() {
 		setEnabled(true);
-		setChecked(feature.getStructure().isHidden());
+		// A selection of features is considered hidden iff every feature is hidden.
```

Comment:
I don't think that every one knows "if and only if" as iff

Reply:
too bad